src/services/base/base.service.js

```diff
@@ -1601,7 +1597,7 @@ module.exports.merge = async (args, {req}, resource_name, collection_name) => {
                     let my_item = newItem[`${i}`];
                     // if newItem[i] does not matches any item in oldItem then insert
                     if (oldItem.every(a => deepEqual(a, my_item) === false)) {
-                        if ('id' in my_item) {
+                        if (typeof my_item === 'object' && 'id' in my_item) {
```

Comment:
just 'object' type will fix the issues here? cos in JS world, pretty much everything is object...

Reply:
so what are we trying to do here? 
to filter cases where this `my_item` is a string, correct?

Reply:
We're really ok with anything that won't error out on the `"id" in my_item` expression. Even if it's something like an Array, that's not really a problem since it won't have an `id` or `sequence` field.

The only exception I see to that rule is `null`, since (very annoyingly) `typeof null === 'object'` in JS. 

I think there are two reasonable ways to handle null. First, we could make it a rule that we always filter null from arrays. Alternatively, we could treat it just like any other value and add a check `my_item !== null` to both if statements. I think the second way is more in line with the currently documented behavior of the merge endpoint, which doesn't specify any special handling of nulls.

Let me know if you have a strong opinion, for now I'm going to go the second route since it matches the documented behavior more closely.

Reply:
ok that makes sense, as long as it matches the behavior, sounds good,
just wanted to check with you. :)